tests/flows/test_fre_flow.py

```diff
@@ -0,0 +1,130 @@
+from ocean_lib.web3_internal.wallet import Wallet
+
+
+def test_fre_flow():
+    config = ExampleConfig.get_config()
+    ocean = Ocean(config)
+
+    # Create Alice's wallet
+    alice_private_key = os.getenv("TEST_PRIVATE_KEY1")
+    alice_wallet = Wallet(
+        ocean.web3,
+        alice_private_key,
+        config.block_confirmations,
+        config.transaction_timeout,
+    )
+    assert alice_wallet.address
+
+    # Mint OCEAN
+    mint_fake_OCEAN(config)
+    assert alice_wallet.web3.eth.get_balance(alice_wallet.address) > 0, "need ETH"
+
+    # Publish an NFT token
+    nft_token = ocean.create_nft_token(
+        "NFTToken1", "NFT1", alice_wallet, "https://oceanprotocol.com/nft/"
+    )
+    token_address = nft_token.address
+    assert token_address
+
+    # Prepare data for ERC20 token
+    erc20_data = ErcCreateData(
+        template_index=1,
+        strings=["Datatoken 1", "DT1"],
+        addresses=[
+            alice_wallet.address,
+            alice_wallet.address,
+            ZERO_ADDRESS,
+            ocean.OCEAN_address,
+        ],
+        uints=[ocean.web3.toWei(200, "ether"), 0],
+        bytess=[b""],
+    )
+
+    tx_id = nft_token.create_erc20(erc20_data, alice_wallet)
+    tx_receipt = ocean.web3.eth.wait_for_transaction_receipt(tx_id)
+
+    erc721_factory = ERC721FactoryContract(
+        ocean.web3, get_address_of_type(config, "ERC721Factory")
+    )
+
+    registered_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_TOKEN_CREATED,
+        tx_receipt.blockNumber,
+        ocean.web3.eth.block_number,
+        None,
+    )
+
+    erc20_address = registered_event[0].args.newTokenAddress
+    print(f"token_address = '{erc20_address}'")
+
+    # Mint the datatokens
+    fixed_price_address = get_address_of_type(config, "FixedPrice")
+    erc20_token = ERC20Token(ocean.web3, erc20_address)
+    erc20_token.mint(alice_wallet.address, to_wei(100), alice_wallet)
+    erc20_token.approve(fixed_price_address, to_wei(100), alice_wallet)
+
+    # Bob buys at fixed rate data tokens
+    bob_private_key = os.getenv("TEST_PRIVATE_KEY2")
+    bob_wallet = Wallet(
+        ocean.web3,
+        bob_private_key,
+        config.block_confirmations,
+        config.transaction_timeout,
+    )
+    print(f"bob_wallet.address = '{bob_wallet.address}'")
+
+    # Verify that Bob has ganache ETH
+    assert ocean.web3.eth.get_balance(bob_wallet.address) > 0, "need ganache ETH"
+
+    OCEAN_token = ERC20Token(ocean.web3, ocean.OCEAN_address)
+
+    # Create exchange_id for a new exchange
+    fixed_rate_data = FixedData(
+        fixed_price_address=fixed_price_address,
+        addresses=[
+            ocean.OCEAN_address,
+            alice_wallet.address,
+            alice_wallet.address,
+            ZERO_ADDRESS,
+        ],
+        uints=[erc20_token.decimals(), OCEAN_token.decimals(), to_wei(1), int(1e15), 0],
+    )
+    tx_id = erc20_token.create_fixed_rate(
```

Comment:
please handle all the approvals, and address retrievals inside the create_fixed_rate_exchange function. Ocean class allready knows about these things. Absolutely ZERO tx handling in readmes. No address retrievals from config directly if we can use Ocean class instead. And preferably as few approvals as possible.